tribe-ext-settings-import-export.php

```diff
@@ -253,76 +233,94 @@ function tribe_sie_process_settings_import() {
-				wp_die( __( 'Please upload a valid .json file.', 'tribe-ext-settings-import-export' ) );
-			}
+				$import_file = $_FILES['import_file']['tmp_name'];
+				$import_filename = $_FILES['import_file']['name'];
+
+				if ( empty( $import_file ) ) {
+					wp_die( __( 'Please upload a file to import.', 'tribe-ext-settings-import-export' ), __( 'Import error', 'tribe-ext-settings-import-export' ), [ 'back_link' => true ] );
+				}
 
-			// Retrieve the settings from the file and convert the json object to an array.
-			$settings = json_decode( file_get_contents( $import_file ), true );
+				if ( ! empty( $import_filename ) ) {
+					$tmp = explode( '.', $import_filename );
+					$extension = end( $tmp );
+				}
 
-			if ( false === $settings ) {
-				wp_die( __( 'Sorry, we could not decode the file.', 'tribe-ext-settings-import-export' ) );
-			}
-			elseif ( ! is_array( $settings ) ) {
-				wp_die( __( 'Sorry, the decoded data is not an array', 'tribe-ext-settings-import-export' ) );
-			}
+				if ( ! isset ( $extension ) || $extension != 'json' ) {
+					wp_die( __( 'Please upload a valid .json file.', 'tribe-ext-settings-import-export' ), __( 'Import error', 'tribe-ext-settings-import-export' ), [ 'back_link' => true ] );
+				}
 
-			if ( update_option( 'tribe_events_calendar_options', $settings ) ) {
-				$action = 'import_success';
-			} else {
-				$action = 'import_failed';
-			}
+				// Retrieve the settings from the file and convert the json object to an array.
+				$settings = json_decode( file_get_contents( $import_file ), true );
 
-			wp_safe_redirect( admin_url( 'edit.php?post_type=tribe_events&page=tribe_import_export&action=' . $action ) );
-			exit;
-		}
+				if ( false === $settings ) {
+					wp_die( __( 'Sorry, we could not decode the file.', 'tribe-ext-settings-import-export' ), __( 'Import error', 'tribe-ext-settings-import-export' ), [ 'back_link' => true ] );
+				}
+				elseif ( ! is_array( $settings ) ) {
+					wp_die( __( 'Sorry, the decoded data is not an array.', 'tribe-ext-settings-import-export' ), __( 'Import error', 'tribe-ext-settings-import-export' ), [ 'back_link' => true ] );
+				}
 
-		/**
-		 * Reset Modern Tribe calendar and ticketing plugins
-		 */
-		function tribe_sie_process_settings_reset() {
+				if ( update_option( 'tribe_events_calendar_options', $settings ) ) {
+					$action = 'import_success';
+				} else {
+					$action = 'import_failed';
+				}
 
-			// Bail if no action
-			if ( empty( $_POST['tribe_sie_action'] ) || 'reset_settings' != $_POST['tribe_sie_action'] ) {
-				return;
+				wp_safe_redirect( admin_url( 'edit.php?post_type=tribe_events&page=tribe_import_export&action=' . $action ) );
 			}
 
-			// Bail if no nonce
-			if ( ! wp_verify_nonce( $_POST['tribe_sie_import_nonce'], 'tribe_sie_import_nonce' ) ) {
-				return;
-			}
+			/**
+			 * Reset actions.
+			 * Reset Modern Tribe calendar and ticketing plugins.
+			 */
+			if ( ! empty ( $_POST['reset'] ) ) {
```

Comment:
Minor/code formatting thing:

```suggestion
			if ( ! empty( $_POST['reset'] ) ) {
```

Reply:
Resolved